### PR TITLE
Move 'new version' support into Platform

### PR DESCRIPTION
**Pull request by dbkr**
* Allow platforms to specify release notes
 * Make web update checking much, much simpler (ie. not manually using XMLHTTPRequest)

### CODE REVIEW

File: src/components/views/globals/NewVersionBar.js [hunk 1]
```diff
@@ -31,35 +32,56 @@ function checkVersion(ver) {
...
     const onChangelogClicked = () => {
-        const ChangelogDialog = sdk.getComponent('dialogs.ChangelogDialog');
-
-        Modal.createDialog(ChangelogDialog, {
-            version: props.version,
-            newVersion: props.newVersion,
-            onFinished: (update) => {
-                if(update) {
-                    window.location.reload();
+        if (props.releaseNotes) {
```
Comment (richvdh) [inline src/components/views/globals/NewVersionBar.js]: nit: I feel like this should be two separate functions

**Comment by richvdh** [comment]
Oh, found it: https://github.com/matrix-org/matrix-react-sdk/pull/532

File: src/components/views/globals/NewVersionBar.js (hunk 1 above)
Comment (richvdh) [inline src/components/views/globals/NewVersionBar.js]: still think this needs splitting

File: src/components/views/globals/NewVersionBar.js [hunk 2]
```diff
@@ -31,35 +32,56 @@ function checkVersion(ver) {
...
+                        PlatformPeg.get().installUpdate();
+                    }
+                }
+            });
+        } else {
+            const ChangelogDialog = sdk.getComponent('dialogs.ChangelogDialog');
+            Modal.createDialog(ChangelogDialog, {
+                version: props.version,
+                newVersion: props.newVersion,
+                releaseNotes: releaseNotes,
```
Comment (richvdh) [inline src/components/views/globals/NewVersionBar.js]: `releaseNotes` seems to be undefined here.

File: src/components/views/globals/NewVersionBar.js [hunk 3]
```diff
@@ -31,35 +32,56 @@ function checkVersion(ver) {
...
+
+    const onUpdateClicked = () => {
+        PlatformPeg.get().installUpdate();
     };
 
-    let changelog_button;
-    if (checkVersion(props.version) && checkVersion(props.newVersion)) {
-        changelog_button = <button className="mx_MatrixToolbar_action" onClick={onChangelogClicked}>Changelog</button>;
+    let action_button;
+    if (props.releaseNotes || (checkVersion(props.version) && checkVersion(props.newVersion))) {
```
Comment (richvdh) [inline src/components/views/globals/NewVersionBar.js]: I could do with a comment here explaining that we show a questiondialog if we have releasenotes, or a changelogdialog if we don't.

**Comment by dbkr** [comment]
ptal

**Pull request merged**